manifest.go

```diff
@@ -156,6 +158,48 @@ func validateManifest(s string) ([]error, error) {
 	return warns, nil
 }
 
+// ValidateProjectRoots validates the project roots present in manifest.
+func ValidateProjectRoots(c *Ctx, m *Manifest, sm gps.SourceManager) error {
+	// Channel to receive all the errors
+	errorCh := make(chan error, len(m.Constraints)+len(m.Ovr))
+
+	var wg sync.WaitGroup
+
+	validate := func(pr gps.ProjectRoot) {
+		defer wg.Done()
+		origPR, err := sm.DeduceProjectRoot(string(pr))
+		if err != nil {
+			errorCh <- err
```

Comment:
Maybe prefix the pr just in case the error message is not clear?

Reply:
the error messages is returned with PR in it already
```
 ✗ github.com/go-yaml is not a valid path for a source on github.com
```
So, it's fine :)